fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java

```diff
@@ -121,46 +123,59 @@ private void handleInteraction(ScenarioTrafficLightRegistration interaction) {
      */
     private void handleInteraction(ScenarioVehicleRegistration scenarioVehicle) throws InternalFederateException {
         if (framework != null) {
-            final CPrototype prototype = framework.getPrototypeByName(scenarioVehicle.getVehicleType().getName());
-            if (prototype == null) {
-                log.debug(
-                        "There is no such prototype \"{}\" configured. No application will be mapped for vehicle \"{}\".",
-                        scenarioVehicle.getVehicleType().getName(),
-                        scenarioVehicle.getId()
-                );
-                return;
-            }
 
-            if (randomNumberGenerator.nextDouble() >= ObjectUtils.defaultIfNull(prototype.weight, 1.0)) {
-                log.debug(
-                        "This scenario vehicle \"{}\" of prototype \"{}\" will not be equipped due to a weight condition of {}.",
-                        scenarioVehicle.getId(),
-                        scenarioVehicle.getVehicleType().getName(),
-                        prototype.weight
-                );
-                return;
+            final List<CPrototype> typeDistribution = framework.getTypeDistributionByName(scenarioVehicle.getVehicleType().getName());
+            if (!typeDistribution.isEmpty()) {
+                final CPrototype selected  = new StochasticSelector<>(typeDistribution, randomNumberGenerator).nextItem();
+                sendVehicleRegistrationForScenarioVehicle(scenarioVehicle, selected.group, selected.applications);
+            } else {
+                final CPrototype prototype = framework.getPrototypeByName(scenarioVehicle.getVehicleType().getName());
+                if (prototype == null) {
```

Comment:
But I do return here after log, or what do you mean?

Reply:
No, there is no need to define a prototype in the `prototype` section if a `typeDistribution` (with entries) is provided. However, it is currently not possible to refer from a typeDistribution entry to a previous defined prototype from the `prototypes` section. This may be a different issue if we want to support this.

Reply:
Okay I understand now!

Reply:
I adjusted this slightly, so now it's possible to refer to a prototype from the `prototypes` section from inside a `typeDistribution`.